Add unlink extension

teaneedz opened the pull request:
## Description

## UnLink - Clean Your Messy URLs

**What it does:**
- Removes tracking parameters (utm_*, fbclid, etc.)
- Extracts clean emails from mailto links  
- Resolves redirects and shortened URLs
- Adds page titles for better readability
- Handles Customer.io and other encoded tracking links

**Why it's useful:**
- Cleans up messy clipboard URLs instantly
- Makes sharing links cleaner and more professional
- Removes privacy-invasive tracking parameters

**How to use:**
Copy any messy URL, run the extension, get a clean URL back in your clipboard!

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

andreaselia commented:
Please could you also read the links in the checklist, as they'll let you know what to do regarding metadata screenshots and the extension icon.

<img width="1200" height="374" alt="CleanShot 2025-07-14 at 22 39 07@2x" src="https://github.com/user-attachments/assets/45fe539b-1f71-4b57-ad97-57bb9876a4e5" />

andreaselia commented:
Feel free to mark the PR as ready for review again when done.

andreaselia commented:
@teaneedz how come you marked this as "ready for review" when no changes have been made to resolve the issues above? 😅

teaneedz commented:
Here's my complete extension with ALL files including metadata and screenshots: https://github.com/teaneedz/unlink-raycast-extension

> On Jul 15, 2025, at 10:11 AM, Andreas Elia ***@***.***> wrote:
> 
> 
> andreaselia
>  left a comment 
> (raycast/extensions#20404)
>  <https://github.com/raycast/extensions/pull/20404#issuecomment-3074479546>
> @teaneedz <https://github.com/teaneedz> how come you marked this as "ready for review" when no changes have been made to resolve the issues above? 😅
> 
> I actually updated and pushed the changes as indicated. Icons created, category added, screenshots added. I also discussed why UnLink should exist next on its own within the extension library due to its unique features. Am I missing something?
> 
> @teaneedz <https://github.com/teaneedz> there's only 1 commit? Did you forget to push new changes?
> 
> ***@***.*** (view on web) <https://github.com/user-attachments/assets/076d038c-6db0-4f08-8c48-042b19aaf813>
> —
> Reply to this email directly, view it on GitHub <https://github.com/raycast/extensions/pull/20404#issuecomment-3074479546>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABHE3HJZ7EQW7NWQFKNGVJL3IUY3HAVCNFSM6AAAAACBMBFUP2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANZUGQ3TSNJUGY>.
> You are receiving this because you were mentioned.
>

teaneedz commented:
@andreaselia I've read your documentation thoroughly and followed every step multiple times.

What I've Done:

Created the extension following your exact guidelines
Added metadata folder with 3 screenshots as required
Fixed memory leaks with proper cleanup
Added categories to package.json
Created dual-mode icons (light/dark)
Updated README formatting
Attempted to upload files through multiple methods including web interface and direct commits
The Issue: Your PR system shows 11 files but cannot detect the metadata folder that clearly exists in my repository. I've tried uploading the complete extension multiple times through different approaches, yet your bot consistently reports "no evidence of metadata folder."

Evidence: My complete extension with all required files: https://github.com/teaneedz/unlink-raycast-extension

This extension provides functionality beyond your existing Link Cleaner (email cleaning, page title fetching, redirect resolution, Customer.io decoding) and meets all technical requirements listed in your checklist.

Request: Please review the extension based on its technical merits rather than infrastructure issues with file detection. If there's a specific technical reason why this extension cannot be accepted, I'd appreciate direct feedback on the code and functionality.

I’ve gone from the terminal to the website method of updating the repo and PR. My side is showing all updated files, from local device to GitHub online public repo.


> On Jul 15, 2025, at 11:58 AM, Andreas Elia ***@***.***> wrote:
> 
> 
> andreaselia
>  left a comment 
> (raycast/extensions#20404)
>  <https://github.com/raycast/extensions/pull/20404#issuecomment-3075069534>
> Hey @teaneedz <https://github.com/teaneedz> 👋
> 
> You might find it helpful to read through the information in each point in the checklist in your main post. There's some really useful stuff there. For example, the 2nd point walks you through how to publish changes to your extension.
> 
> Here's the full guide if you'd like to dive deeper:
> https://developers.raycast.com/basics/publish-an-extension
> 
> Feel free to reach out if you run into any issues.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/raycast/extensions/pull/20404#issuecomment-3075069534>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABHE3HJJCL3NRYXMHUXDZWT3IVFODAVCNFSM6AAAAACBMBFUP2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANZVGA3DSNJTGQ>.
> You are receiving this because you were mentioned.
>

andreaselia commented:
Hi @teaneedz 👋 

Extensions must be published to this repository, we only review code submitted as part of the extension PR.

Have you tried running the `npm run publish` command to update this PR with your changes?

https://developers.raycast.com/basics/publish-an-extension#publish-your-extension

andreaselia commented:
Nice work on getting your changes published into the PR @teaneedz 👍 

I left a comment above regarding indentation, and please re-take the screenshots for the metadata, as they show old icons.

<img width="862" height="1538" alt="CleanShot 2025-07-16 at 08 41 48@2x" src="https://github.com/user-attachments/assets/51aefb6d-4768-44fe-aae4-5703bcf46ae8" />

andreaselia commented:
@teaneedz did you use the built-in screenshot tool? Your screenshots look offset and the Raycast window isn't central. You can see this more so in the `extensions/unlink/metadata/unlink-cleaned-message.png` screenshot.

You can read more here: https://developers.raycast.com/basics/prepare-an-extension-for-store#adding-screenshots

andreaselia commented:
@teaneedz we only allow screenshots taken with the tool as it avoids issues like too much space on one side of a screenshot.

Could you post the issue you were having in the form of a screen recording if possible? Were you able to follow the documentation for it?
